lib/galaxy/webapps/galaxy/api/__init__.py

```diff
@@ -150,11 +150,22 @@ def __call__(self, name: str, **path_params):
 DependsOnUser = Depends(get_user)
 
 
+def get_current_history_from_session(galaxy_session: Optional[model.GalaxySession]) -> Optional[model.History]:
+    if galaxy_session:
+        return galaxy_session.current_history
+    return None
+
+
 def get_trans(request: Request, app: StructuredApp = DependsOnApp, user: Optional[User] = Depends(get_user),
               galaxy_session: Optional[model.GalaxySession] = Depends(get_session),
               ) -> SessionRequestContext:
     url_builder = UrlBuilder(request)
-    return SessionRequestContext(app=app, user=user, galaxy_session=galaxy_session, url_builder=url_builder, host=request.client.host)
+    return SessionRequestContext(
+        app=app, user=user,
+        galaxy_session=galaxy_session,
+        url_builder=url_builder, host=request.client.host,
+        history=get_current_history_from_session(galaxy_session),
```

Comment:
The `current_history` seemed to be always None when using Galaxy as an anonymous user in FastAPI mode.

Hopefully getting it from the session is the correct way to set it, right?

Reply:
Yes, I think that looks right